test/lib/deployment/test-deployment.js

```diff
@@ -220,26 +220,47 @@ async function runProbe(probe, deploymentId, deploymentUrl, ctx) {
     hadTest = true;
   }
 
+  /**
+   * @type Record<string, string[]>
+   */
+  const rawHeaders = resp.headers.raw();
   if (probe.responseHeaders) {
     // eslint-disable-next-line no-loop-func
     Object.keys(probe.responseHeaders).forEach(header => {
-      const actual = resp.headers.get(header);
-      const expected = probe.responseHeaders[header];
-      const isEqual = Array.isArray(expected)
-        ? expected.every(h => actual.includes(h))
-        : typeof expected === 'string' &&
-          expected.startsWith('/') &&
-          expected.endsWith('/')
-        ? new RegExp(expected.slice(1, -1)).test(actual)
-        : expected === actual;
-      if (!isEqual) {
-        const headers = Array.from(resp.headers.entries())
-          .map(([k, v]) => `  ${k}=${v}`)
-          .join('\n');
+      const actualArr = rawHeaders[header.toLowerCase()];
```

Comment:
Are raw headers always lowercase keys?

Reply:
They are not.

Reply:
So why is the `header.toLowerCase()` there? Are we assuming the server will always lowercase the key before sending the response?

Reply:
I guess so. Some existing tests were using lowercase.